src/main/java/org/cloudfoundry/promregator/cfaccessor/ReactiveCFAccessorImpl.java

```diff
@@ -377,6 +379,13 @@ private void setupPaginatedRequestFetcher() {
 		return this.paginatedRequestFetcher.performGenericRetrieval(RequestType.SPACE_SUMMARY, spaceId, 
 				request, r -> this.cloudFoundryClient.spaces().getSummary(r), this.requestTimeoutAppSummary);
 
+  }
+  
+	@Override
+	public Mono<ListDomainsResponse> retrieveDomains() {
+		ListDomainsRequest request = ListDomainsRequest.builder().build();
+		
+		return this.paginatedRequestFetcher.performGenericRetrieval(RequestType.DOMAINS, "(empty)", 
+				request, r -> this.cloudFoundryClient.domainsV3().list(request), this.requestTimeoutAppSummary);
```

Comment:
You are mixing V2 and V3 interfaces here.
At ours, we still must assume that we only have v2 API endpoints available.

Reply:
The domain api for v2 is marked as deprecated and causes compiler warnings. I dont think the rest of the v2 API does this however and im not sure why.